vcd/api_test.go

```diff
@@ -72,3 +84,77 @@ func TestTags(t *testing.T) {
 		showTags()
 	}
 }
+
+// Finds the current directory, through the path of this running test
+func getCurrentDir() string {
+	_, currentFilename, _, _ := runtime.Caller(0)
+	return filepath.Dir(currentFilename)
+}
+
+// Reads the version from the VERSION file in the root directory
+func getMajorVersion() string {
+
+	versionFile := path.Join(getCurrentDir(), "..", "VERSION")
+
+	// Checks whether the VERSION file exists
+	_, err := os.Stat(versionFile)
+	if os.IsNotExist(err) {
+		panic("Could not find VERSION file")
+	}
+
+	// Reads the version from the file
+	versionText, err := ioutil.ReadFile(versionFile)
+	if err != nil {
+		panic(fmt.Errorf("could not read VERSION file %s: %v", versionFile, err))
+	}
+
+	// The version is expected to be in the format v#.#.#
+	// We only need the first two numbers
+	reVersion := regexp.MustCompile(`v(\d+\.\d+)\.\d+`)
+	versionList := reVersion.FindAllStringSubmatch(string(versionText), -1)
+	if versionList == nil || len(versionList) == 0 {
+		panic("empty or non-formatted version found in VERSION file")
+	}
+	if versionList[0] == nil || len(versionList[0]) < 2 {
+		panic("unable to extract major version from VERSION file")
+	}
+	// A successful match will look like
+	// [][]string{[]string{"v2.0.0", "2.0"}}
+	// Where the first element is the full text matched, and the second one is the first captured text
+	return versionList[0][1]
+}
+
+// Checks that the provider header in index.html.markdown
+// has the version defined in the VERSION file
+func TestProviderVersion(t *testing.T) {
+	indexFile := path.Join(getCurrentDir(), "..", "website", "docs", "index.html.markdown")
+	_, err := os.Stat(indexFile)
+	if os.IsNotExist(err) {
+		fmt.Printf("%s\n", indexFile)
+		panic("Could not find index.html.markdown file")
+	}
+
+	indexText, err := ioutil.ReadFile(indexFile)
+	if err != nil {
+		panic(fmt.Errorf("could not read index file %s: %v", indexFile, err))
+	}
+
+	vcdHeader := `# VMware vCloud Director Provider`
+	expectedText := vcdHeader + ` ` + currentProviderVersion
+	reExpectedVersion := regexp.MustCompile(`(?m)^` + expectedText)
+	reFoundVersion := regexp.MustCompile(`(?m)^` + vcdHeader + ` \d+\.\d+`)
+	if reExpectedVersion.MatchString(string(indexText)) {
+		t.Logf("Found expected version <%s> in index.html.markdown", currentProviderVersion)
```

Comment:
I'd say we don't need to be too verbose when expected version is found:
```
=== RUN   TestProviderVersion
--- PASS: TestProviderVersion (0.00s)
    api_test.go:147: Found expected version <2.2> in index.html.markdown
```

Reply:
Normally, I would let the test to pass silently. In this case I think we should see the version at every test run, to make us aware of what we are running with.

Alternatively, I can show this message depending on an environment variable. I can't tie the message to the test verbosity, because it is mandatory for Terraform test.

Reply:
I have added an environment variable to show details.
The test is now mute by default.

Reply:
Thanks, TESTING.md needs updates about the new unit group and addition for the new env variable.

Reply:
TESTING.md updated